### PR TITLE
I3C Base support

### PR DESCRIPTION
**Pull request by ojayanth**
- I3C code sync from aspeed tree.
- Added DTS support AMD onyx CRB for both I2C and I3C mode.
- Added patches to enable APML sbrmi and sbtsi driver.

### CODE REVIEW

File: arch/arm/boot/dts/aspeed/aspeed-g6.dtsi [hunk 1]
```diff
@@ -589,6 +595,13 @@
 				status = "disabled";
 			};
 
+			i3c: bus@1e7a0000 {
```
Comment (mahkurap) [inline arch/arm/boot/dts/aspeed/aspeed-g6.dtsi:598]: not sure what this is needed for.

File: arch/arm/boot/dts/aspeed/aspeed-g6.dtsi [hunk 2]
```diff
@@ -1146,3 +1159,163 @@
...
+
+                reg = <0x2000 0x1000>;
+                compatible = "aspeed,ast2600-i3c";
+                clocks = <&syscon ASPEED_CLK_GATE_I3C0CLK>;
+                resets = <&syscon ASPEED_RESET_I3C0>;
+                interrupts = <GIC_SPI 102 IRQ_TYPE_LEVEL_HIGH>;
+                pinctrl-names = "default";
+                pinctrl-0 = <&pinctrl_i3c1_default>;
+
+                i2c-scl-hz = <1000000>;
```
Comment (mahkurap) [inline arch/arm/boot/dts/aspeed/aspeed-g6.dtsi:1183]: for Turin platforms we have i2c at 1MHz, and i3c at 12.5MHz int the dtsi file. i3c seems to be at a different frequency. is it this taken from the upstream as-is?

File: arch/arm/boot/dts/aspeed/ast2600-amd-bmc-common.dtsi [hunk 3]
```diff
@@ -0,0 +1,568 @@
+// SPDX-License-Identifier: GPL-2.0+
```
Comment (mahkurap) [inline arch/arm/boot/dts/aspeed/ast2600-amd-bmc-common.dtsi:1]: Are we sure the below device configuration is common across all the Venice platforms? What do we do if it is different? We will overwrite it in the platform specific dts file?

File: arch/arm/boot/dts/aspeed/aspeed-bmc-amd-onyx.dts [hunk 4]
```diff
@@ -26,3 +26,98 @@
 		reg = <0x4c>;
 	};
 };
+
+#include <dt-bindings/i3c/i3c.h>
+
+#define JESD300_SPD_I2C_MODE(bus, index, addr) \
```
Comment (supven01) [inline arch/arm/boot/dts/aspeed/aspeed-bmc-amd-onyx.dts:32]: As discussed, please modify it in a manner that will be helpful for DT overlay in the future.

**Review by mahkurap — APPROVED** [review]

**Review by mahkurap — APPROVED** [review]

**Review by mahkurap — APPROVED** [review]

**Review by mahkurap — APPROVED** [review]

**Review by mahkurap — APPROVED** [review]

**Review by mahkurap — APPROVED** [review]

**Review by mahkurap — APPROVED** [review]

File: arch/arm/boot/dts/aspeed/ast2600-amd-bmc-common.dtsi (hunk 3 above)
Comment (supven01) [reply arch/arm/boot/dts/aspeed/ast2600-amd-bmc-common.dtsi:1]: This is applicable for Turin/Genoa with AST2600 boards only. For Venice w/ AST2700 boards, a new file will be created.

File: arch/arm/boot/dts/aspeed/aspeed-bmc-amd-onyx.dts (hunk 4 above)
Comment (supven01) [reply arch/arm/boot/dts/aspeed/aspeed-bmc-amd-onyx.dts:32]: Temporary change. Subject to change after DT overlay is implemented.

**Comment by ojayanth** [comment]
As discussed remove Marley Chages now , Once APML chnages part verified , will create new PR to enable Marley APML.

File: arch/arm/boot/dts/aspeed/aspeed-g6.dtsi [hunk 5]
```diff
@@ -1146,3 +1159,163 @@
...
+                compatible = "aspeed,ast2600-i3c";
+                clocks = <&syscon ASPEED_CLK_GATE_I3C4CLK>;
+                resets = <&syscon ASPEED_RESET_I3C4>;
+                interrupts = <GIC_SPI 106 IRQ_TYPE_LEVEL_HIGH>;
+                pinctrl-names = "default";
+                pinctrl-0 = <&pinctrl_i3c5_default>;
+
+                i2c-scl-hz = <1000000>;
+                i3c-scl-hz = <2000000>;
+                i3c-od-scl-hi-period-ns = <380>;
```
Comment (rajaganeshr) [inline arch/arm/boot/dts/aspeed/aspeed-g6.dtsi:1285]: How did we arrive at these values for hi and lo?

Comment (ojayanth) [reply arch/arm/boot/dts/aspeed/aspeed-g6.dtsi:1285]: This is copy of aspeed version.

Comment (rajaganeshr) [reply arch/arm/boot/dts/aspeed/aspeed-g6.dtsi:1285]: Ok, in driver we may need to override tx hold and timing delay later.

#define SDA_DLY_TIMING_OFFSET        0xD0
#define SDA_TX_HOLD_APML             0x00030000
#define SDA_TX_HOLD_DIMM             0x00070000

**Review by rajaganeshr — APPROVED** [review]

**Pull request merged**